### PR TITLE
[HPRO-917] Adjust routing rules for `/s/cron` endpoints

### PR DESCRIPTION
**Pull request by stephenyeargin**
| Q                   | A
| ------------------- | --------------
| Target Release?     | Next available 
| Bug fix?            | ✅ 
| New feature?        | ❌ 
| Database migration? | ❌ 
| New configuration?  | ❌ 
| Composer updates?   | ✅ 
| NPM updates?        | ❌ 
| Jira Ticket(s)      | HPRO-917 

### Summary

Adjusts routing rules around `/s/cron` endpoints.

### Testing

Provide the `X-Appengine-Cron: true` header when interacting with the endpoints. Omitting it will return a 404 error.

Example:

```
$ curl -H 'X-Appengine-Cron: true' http://localhost:8080/s/cron/ping-test
```

### CODE REVIEW

File: symfony/src/Controller/CronController.php [hunk 1]
```diff
@@ -20,7 +20,7 @@
 use Symfony\Component\Routing\Annotation\Route;
 
 /**
- * @Route("/s/cron")
+ * @Route("/s/cron", condition="request.headers.get('X-Appengine-Cron') matches '/true/'")
```
Comment (jt2k) [inline symfony/src/Controller/CronController.php:23]: What's the reason for a regex match here? Do we not want the exact string 'true'?

**Comment by shyamnaru** [comment]
Previously we are not running this check locally https://github.com/all-of-us/healthpro/blob/0729b0ea34964668c602d676d30f689b91e137bd/src/Pmi/Controller/CronController.php#L19-L22. This change will prevent hitting the cron urls from the local browser. As you mentioned we can use curl to run the cron jobs.

File: symfony/src/Controller/CronController.php (hunk 1 above)
Comment (stephenyeargin) [reply symfony/src/Controller/CronController.php:23]: Example code only used `matches`, so I'd assumed that it had to be regular expression. Further reading indicates I can get away with an exact string. (`==` or `===`)

**Comment by stephenyeargin** [comment]
>  This change will prevent hitting the cron urls from the local browser. As you mentioned we can use curl to run the cron jobs.

I'd considered this as well. Because it's particularly important that we make sure this operates behind the check, I didn't want to make an exception for local development. There are a handful of plugins to modify the request headers your browser sends to continue to be able to test calls that way.

Example: https://github.com/bewisse/modheader

File: symfony/src/Controller/CronController.php (hunk 1 above)
Comment (stephenyeargin) [reply symfony/src/Controller/CronController.php]: Resolved in 6c125e4.

**Review by jt2k — APPROVED** [review]

**Review by shyamnaru — APPROVED** [review]

**Pull request merged**